### PR TITLE
Initial changes to the egads code base

### PR DESCRIPTION
**Pull request by ianflint**
Models are no longer dependent on global Storage class
Converted logs to use log4j
Replaced interfaces with abstract base classes and pulled common members and methods into base classes
Defined new output types
   STDOUT
   STDOUT_ALL
   PLOT
Eliminated unused variables and imports
Commented out some unused methods (there was some work put into them so I didn't delete them.)

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by ianflint** [comment]
I just signed my CLA

**Comment by yahoocla** [comment]
CLA is valid!

File: pom.xml [hunk 1]
```diff
@@ -22,6 +21,21 @@
     <dependencies>
         <!-- needed by egads -->
         <dependency>
+          <groupId>com.fasterxml.jackson.core</groupId>
+          <artifactId>jackson-databind</artifactId>
+          <version>2.4.4</version>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.logging.log4j</groupId>
+          <artifactId>log4j-api</artifactId>
```
Comment (lbordowitz) [inline pom.xml]: Not necessary; the log4j-api artifact is a dependency of the log4j-core artifact.

File: src/main/java/com/yahoo/egads/models/adm/AdaptiveKernelDensityChangePointDetector.java [hunk 2]
```diff
@@ -156,10 +154,10 @@ public IntervalSequence detect(DataSequence observedSeries, DataSequence expecte
 
         // Detecting change points
         ArrayList<Integer> changePoints =
-                        detectChnagePoints(residuals, this.preWindowSize, this.postWindowSize, this.confidence);
+                        detectChangePoints(residuals, this.preWindowSize, this.postWindowSize, this.confidence);
 
         // Preparing the output
-        if (Storage.debug == 3) {
+        if (this.outputDest.equals("STD_OUT_ALL")) {
```
Comment (lbordowitz) [inline src/main/java/com/yahoo/egads/models/adm/AdaptiveKernelDensityChangePointDetector.java]: log4j log levels should take care of this.

File: src/main/java/com/yahoo/egads/models/adm/AnomalyDetectionAbstractModel.java [hunk 3]
```diff
@@ -7,13 +7,28 @@
...
 import org.json.JSONObject;
 import org.json.JSONStringer;
+
 import com.yahoo.egads.data.JsonEncoder;
 
-public abstract class AnomalyDetectionAbstractModel implements
-        AnomalyDetectionModel {
+public abstract class AnomalyDetectionAbstractModel implements AnomalyDetectionModel {
+
+	protected org.apache.logging.log4j.Logger logger;
```
Comment (lbordowitz) [inline src/main/java/com/yahoo/egads/models/adm/AnomalyDetectionAbstractModel.java]: Import the Logger class.

**Pull request merged**